File: tb3_stop.py
import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist


class TB3Stop(Node):
    def __init__(self) -> None:
        super().__init__("tb3_stop")
        self.publisher_ = self.create_publisher(Twist, "/cmd_vel", 10)
        self.timer = self.create_timer(0.5, self.timer_callback)

    def timer_callback(self) -> None:
        msg = Twist()
        self.publisher_.publish(msg)
        self.get_logger().info("Publishing stop command...")


def main(args=None) -> None:
    rclpy.init(args=args)
    node = TB3Stop()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == "__main__":
    main()

File: tb3_stop.cpp
#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/twist.hpp"

class TB3Stop : public rclcpp::Node
{
public:
  TB3Stop() : Node("tb3_stop")
  {
    publisher_ =
      this->create_publisher<geometry_msgs::msg::Twist>("/cmd_vel", 10);

    timer_ = this->create_wall_timer(
      std::chrono::milliseconds(500),
      std::bind(&TB3Stop::timer_callback, this));
  }

private:
  void timer_callback()
  {
    geometry_msgs::msg::Twist msg;

    publisher_->publish(msg);

    RCLCPP_INFO(this->get_logger(), "Publishing stop command...");
  }

  rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<TB3Stop>());
  rclcpp::shutdown();
  return 0;
}